pkg/storagev2/repository.go

```diff
@@ -12,31 +12,38 @@ import (
 
 // Repository is a collection of stores that provides a unified interface
 // to access them. Access to all stores can be guarded by a transaction.
-type Repository struct {
+type Repository interface {
```

Comment:
What is the purpose of having a `Repository` interface?

Reply:
So in the localstore I felt like we should have some tests related to errors. In order to achieve that I had to override the repository in the DB pointer. As the repository is a pointer it is not possible to extend it and change behavior.

Check the new tests in localstore_test.go related to rollbacks.